tests/Unit/Parallel/PhaseControl/Test_PhaseControlTags.cpp

```diff
@@ -0,0 +1,138 @@
+};
+}  // namespace Tags
+
+template <size_t val, typename PhaseChangeRegistrars>
+struct TestCreatable : public PhaseChange<PhaseChangeRegistrars> {
+
+  TestCreatable() = default;
+  explicit TestCreatable(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(TestCreatable);  // NOLINT
+
+  static std::string name() noexcept {
+    if constexpr(val == 1_st) {
+      return "TestCreatable(1)";
+    } else if constexpr(val == 2_st) {
+      return "TestCreatable(2)";
+    } else {
+      return "TestCreatable(Unknown)";
+    }
+  }
+
+  template <typename Metavariables>
+  using participating_components = tmpl::list<>;
+
+  static constexpr Options::String help{"Creatable for test"};
+
+  struct IntOption {
+    using type = int;
+    static constexpr Options::String help{"Option paramer"};
+  };
+
+  using options = tmpl::list<IntOption>;
+  using phase_change_tags_and_combines = tmpl::conditional_t<
+      1_st < val, tmpl::list<Tags::DummyDecisionTag1>,
+      tmpl::list<Tags::DummyDecisionTag2>>;
+
+  explicit TestCreatable(int option_value) noexcept
+      : option_value_{option_value} {}
+
+  void pup(PUP::er& p) noexcept override { p | option_value_; }  // NOLINT
+
+  int option_value_ = 0;
+};
+
+template <size_t val, typename PhaseChangeRegistrars>
+PUP::able::PUP_ID TestCreatable<val, PhaseChangeRegistrars>::my_PUP_ID = 0;
+
+SPECTRE_TEST_CASE("Unit.Parallel.PhaseControl.PhaseControlTags",
+                  "[Unit][Parallel]") {
+  using phase_changes = tmpl::list<Registrars::TestCreatable<1_st>,
+                                   Registrars::TestCreatable<2_st>>;
+  using triggers = tmpl::list<Triggers::Registrars::Slabs>;
+  Parallel::register_derived_classes_with_charm<PhaseChange<phase_changes>>();
+  Parallel::register_derived_classes_with_charm<Trigger<triggers>>();
+  Parallel::register_derived_classes_with_charm<TimeSequence<size_t>>();
+
+  TestHelpers::db::test_simple_tag<
+      PhaseControl::Tags::PhaseChangeAndTriggers<phase_changes, triggers>>(
+      "PhaseChangeAndTriggers");
+
+  const auto created_phase_changes = TestHelpers::test_creation<
+      typename PhaseControl::OptionTags::PhaseChangeAndTriggers<phase_changes,
+                                                                triggers>::type,
+      PhaseControl::OptionTags::PhaseChangeAndTriggers<phase_changes,
+                                                       triggers>>(
+      " - - Slabs:\n"
+      "       EvenlySpaced:\n"
+      "         Interval: 2\n"
+      "         Offset: 0\n"
+      "   - - TestCreatable(1):\n"
+      "         IntOption: 4\n"
+      "     - TestCreatable(2):\n"
+      "         IntOption: 2");
+  CHECK(created_phase_changes.size() == 1_st);
+  const auto& first_creatable = created_phase_changes[0].second[0];
+  const auto& second_creatable = created_phase_changes[0].second[1];
+  CHECK(dynamic_cast<TestCreatable<1_st, phase_changes>*>(
```

Comment:
CHECK -> REQUIRE.  (because the next statement will segfault if the conditional fails)